daemon/daemon_utils.go

```diff
@@ -52,3 +59,93 @@ func initialRuntime(baseDir string, runtimes map[string]types.Runtime) error {
 
 	return nil
 }
+
+// checkSnapshotter checks whether the given snapshotter is valid
+func checkSnapshotter(snapshotter string, containerdRootDir string) error {
+	var (
+		driverFound = false
+	)
+
+	fInfo, err := os.Stat(containerdRootDir)
+	if err != nil {
+		return fmt.Errorf("failed to stat containerd root dir: %v", err)
+	}
+
+	if !fInfo.IsDir() {
+		return fmt.Errorf("containerd root dir %s is not a directory", containerdRootDir)
+	}
+
+	infos, err := ioutil.ReadDir(containerdRootDir)
+	if err != nil {
+		return fmt.Errorf("failed to read dir %s: %v", containerdRootDir, err)
+	}
+
+	// read all snapshotter dir to make sure snapshotter is valid
+	for _, info := range infos {
+		found, err := checkSnapshotterDir(snapshotter, containerdRootDir, info)
+		if err != nil {
+			return err
+		}
+
+		if found {
+			driverFound = found
+		}
+	}
+
+	if !driverFound {
+		return fmt.Errorf("containerd not support snapshotter driver %s", snapshotter)
+	}
+
+	return nil
+}
+
+func checkSnapshotterDir(snapshotter string, containerdRootDir string, info os.FileInfo) (driverFound bool, err error) {
```

Comment:
Could you add some unit test for this function? @wangforthinker

Reply:
Add test for checkSnapshotter function.

Reply:
thx